pkg/broker/ingress/stats_reporter.go

```diff
@@ -0,0 +1,136 @@
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics"
+	"knative.dev/pkg/metrics/metricskey"
+)
+
+// stats_exporter is adapted from knative.dev/eventing/pkg/broker/ingress/stats_reporter.go
+// with the following changes:
+// - Metric descriptions are updated to match GCP broker specifics.
+// - Removed StatsReporter interface and directly use helper methods instead.
+
+var (
+	// dispatchTimeInMsecM records the time spent dispatching an event to
+	// a decouple queue, in milliseconds.
+	dispatchTimeInMsecM = stats.Float64(
+		"event_dispatch_latencies",
+		"The time spent dispatching an event to the decouple topic",
+		stats.UnitMilliseconds,
+	)
+
+	// Create the tag keys that will be used to add tags to our measurements.
+	// Tag keys must conform to the restrictions described in
+	// go.opencensus.io/tag/validate.go. Currently those restrictions are:
+	// - length between 1 and 255 inclusive
+	// - characters are printable US-ASCII
+	namespaceKey         = tag.MustNewKey(metricskey.LabelNamespaceName)
+	brokerKey            = tag.MustNewKey(metricskey.LabelBrokerName)
+	eventTypeKey         = tag.MustNewKey(metricskey.LabelEventType)
+	responseCodeKey      = tag.MustNewKey(metricskey.LabelResponseCode)
+	responseCodeClassKey = tag.MustNewKey(metricskey.LabelResponseCodeClass)
+	podKey               = tag.MustNewKey(metricskey.PodName)
+	containerKey         = tag.MustNewKey(metricskey.ContainerName)
+)
+
+type reportArgs struct {
+	namespace    string
+	broker       string
+	eventType    string
+	responseCode int
+}
+
+func init() {
+	register()
+}
+
+func register() {
+	tagKeys := []tag.Key{
+		namespaceKey,
+		brokerKey,
+		eventTypeKey,
+		responseCodeKey,
+		responseCodeClassKey,
+		podKey,
+		containerKey,
+	}
+
+	// Create view to see our measurements.
+	err := view.Register(
+		&view.View{
+			Name:        "event_count",
+			Description: "Number of events received by a Broker",
+			Measure:     dispatchTimeInMsecM,
+			Aggregation: view.Count(),
+			TagKeys:     tagKeys,
+		},
+		&view.View{
+			Name:        dispatchTimeInMsecM.Name(),
+			Description: dispatchTimeInMsecM.Description(),
+			Measure:     dispatchTimeInMsecM,
+			Aggregation: view.Distribution(metrics.Buckets125(1, 10000)...), // 1, 2, 5, 10, 20, 50, 100, 500, 1000, 5000, 10000
+			TagKeys:     tagKeys,
+		},
+	)
+	if err != nil {
+		log.Fatalf("failed to register opencensus views, %s", err)
+	}
+}
+
+// NewStatsReporter creates a new StatsReporter.
+func NewStatsReporter(podName, containerName string) *StatsReporter {
+	return &StatsReporter{
+		podName:       podName,
+		containerName: containerName,
+	}
+}
+
+// StatsReporter reports ingress metrics.
+type StatsReporter struct {
+	podName       string
+	containerName string
+}
+
+func (r *StatsReporter) reportEventDispatchTime(ctx context.Context, args reportArgs, d time.Duration) error {
+	tag, err := tag.New(
```

Comment:
If the tags don't need to be stored in ctx, just use RecordWithTags instead.

Reply:
This is using `"knative.dev/pkg/metrics"`, which doesn't have `RecordWithTags`. `"knative.dev/pkg/metrics"` manages the metric config.